cmd/namespace_registry.go

```diff
@@ -0,0 +1,253 @@
+
+	req, err := http.NewRequest(method, url, bytes.NewBuffer(payload))
+	if err != nil {
+		panic(err)
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
+
+	body, _ := ioutil.ReadAll(resp.Body)
+	return body
+}
+
+func resp_to_json(body []byte) (map[string]string) {
+	// Unmarshal the response body
+	var respData map[string]string
+	err := json.Unmarshal(body, &respData)
+	if err != nil {
+		panic(err)
+	}
+	return respData
+}
+
+func namespace_register_with_identity(publicKeyPath string, privateKeyPath string, namespaceRegistryEndpoint string, prefix string) () {
+	data := map[string]interface{}{
+		"identity_required": "true",
+	}
+	resp := make_request(namespaceRegistryEndpoint, "POST", data)
+	respData := resp_to_json(resp)
+
+	verification_url := respData["verification_url"]
+	device_code := respData["device_code"]
+	fmt.Printf("Verification URL: %s\n", verification_url)
+
+	done := false
+	for !done {
+		data = map[string]interface{}{
+			"identity_required": "true",
+			"device_code": device_code,
+		}
+		resp = make_request(namespaceRegistryEndpoint, "POST", data)
+		respData = resp_to_json(resp)
+
+		if respData["status"] == "APPROVED" {
+			done = true
+		} else {
+			fmt.Printf("Waiting for approval...\n")
+			reader := bufio.NewReader(os.Stdin)
+			fmt.Print("Press Enter after verification")
+			_, _ = reader.ReadString('\n')
+			// time.Sleep(60 * time.Second)
+		}
+	}
+	access_token := respData["access_token"]
+	fmt.Printf("Access token: %s\n", access_token)
+	namespace_register(publicKeyPath, privateKeyPath, namespaceRegistryEndpoint, access_token, prefix)
+
+}
+
+func namespace_register(publicKeyPath string, privateKeyPath string, namespaceRegistryEndpoint string, access_token string, prefix string) () {
+	publicKey, err := loadPublicKey(publicKeyPath)
+	if err != nil {
+		fmt.Printf("Failed to load public key: %v\n", err)
+		os.Exit(1)
+	}
+
+	privateKey, err := loadPrivateKey(privateKeyPath)
+	if err != nil {
+		fmt.Printf("Failed to load private key: %v\n", err)
+		os.Exit(1)
+	}
+
+	client_nonce, err := generateNonce()
+	if err != nil {
+		fmt.Printf("Failed to generate nonce: %v\n", err)
+		os.Exit(1)
+	}
+
+	data := map[string]interface{}{
+		"client_nonce": client_nonce,
+		"pubkey":   fmt.Sprintf("%x", publicKey.X.Bytes()),
+	}
+
+	resp := make_request(namespaceRegistryEndpoint, "POST", data)
+	respData := resp_to_json(resp)
+
+	// Create client payload by concatenating client_nonce and server_nonce
+	clientPayload := client_nonce + respData["server_nonce"]
+
+	// Sign the payload
+	signature, err := signPayload([]byte(clientPayload), privateKey)
+	if err != nil {
+		panic(err)
+	}
+
+	// Create data for the second POST request
+	data2 := map[string]interface{}{
+		"client_nonce":      client_nonce,
+		"server_nonce":      respData["server_nonce"],
+		"pubkey":        	 map[string]string{
```

Comment:
May be possible to replace this with a JWT library call as well.

Reply:
✅